force-app/main/default/classes/ProgramEngagementSelector.cls

```diff
@@ -36,35 +56,30 @@ public with sharing class ProgramEngagementSelector {
         List<ProgramEngagement__c> programEngagements = [
             SELECT
                 Id,
+                Contact__c,
                 Contact__r.Name,
                 Contact__r.Email,
                 Stage__c,
                 Program__r.Name,
                 ProgramCohort__c
             FROM ProgramEngagement__c
-            WHERE
-                Stage__c IN :engagementStage
-                AND Program__c IN (
-                    SELECT Program__c
-                    FROM Service__c
-                    WHERE Id = :serviceId
-                )
+            WHERE Stage__c IN :engagementStage AND Program__c = :programId
```

Comment:
We needed the program to be returned in the model so since we had the program, it made sense to me to simplify the query. It is also more optimal as you mentioned as well.